// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// Represents the most recent snapshot of a thread when a user is interacting
// with AI features.
message AiThreadSpecifics {
  enum ThreadType {
    UNKNOWN = 0;
    AI_MODE = 1;
  }

  // Type of the thread.
  optional ThreadType type = 1;

  // Unique identifier for the thread.
  optional string server_id = 2;

  // Latest encrypted state token of the conversation.
  optional string conversation_turn_id = 3;

  // Title of the thread.
  optional string title = 4;
}
